File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/application/book/query/GetBookService.java
package dev.starryeye.book_library.application.book.query;

import dev.starryeye.book_library.domain.book.BookRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional(readOnly = true)
@Service
@RequiredArgsConstructor
public class GetBookService {

    private final BookRepository bookRepository;

    public boolean existsBy(Long bookId) {
        return bookRepository.existsById(bookId);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/book/query/GetBookService.kt
package dev.starryeye.book_library.application.book.query

import dev.starryeye.book_library.domain.book.BookRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional(readOnly = true)
@Service
class GetBookService(
    private val bookRepository: BookRepository,
) {

    fun existsBy(bookId: Long): Boolean {
        return bookRepository.existsById(bookId)
    }
}